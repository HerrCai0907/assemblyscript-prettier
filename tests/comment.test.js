import * as prettier from "prettier";
import asPlugin from "../src/plugin.js";

test("block comment", async () => {
  const originCode = `
/*1*/ 
@lazy/*2*/ let offset = 1;
/*3*/
@lazy /*4*/let offset = 1;
/*5*/
`;

  const formatedCode = await prettier.format(originCode, {
    parser: "typescript",
    plugins: [asPlugin],
  });

  expect(formatedCode).toMatchSnapshot();
});

test("line comment", async () => {
  const originCode = `
// 1
  @lazy
  //2
   let offset = 1;
// 3
`;

  const formatedCode = await prettier.format(originCode, {
    parser: "typescript",
    plugins: [asPlugin],
  });

  expect(formatedCode).toMatchSnapshot();
});
