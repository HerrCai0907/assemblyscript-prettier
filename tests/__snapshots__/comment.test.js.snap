// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block comment 1`] = `
"/*1*/
@lazy /*2*/ let offset = 1;
/*3*/
@lazy /*4*/ let offset = 1;
/*5*/
"
`;

exports[`line comment 1`] = `
"// 1
@lazy
//2
let offset = 1;
// 3
"
`;
